http.go

```diff
@@ -88,6 +89,10 @@ func watchHandler(response http.ResponseWriter, req *http.Request, list *memberl
 
 	var jsonBytes []byte
 	pushUpdate := func() error {
+		// Enter critical section
+		state.RLock()
+		defer state.RUnlock()
```

Comment:
Are you sure you want to hold the lock this long @mihaitodor?  It seems to me you could get away with releasing it after the if/else block.